src/tron/TronEngine.ts

```diff
@@ -1422,14 +1365,8 @@ export class TronEngine extends CurrencyEngine<TronTools, SafeTronWalletInfo> {
 
     const unstakeNativeAmount = sub(claimedAmount, networkFee)
     const edgeTransaction: EdgeTransaction = {
-      action: {
-        type: 'unstake',
-        stakeAssets: [
-          {
-            pluginId: this.currencyInfo.pluginId,
-            nativeAmount: unstakeNativeAmount
-          }
```

Comment:
We split `action` into `assetAction` and `chainAction`. The `chainAction` represents the total effect of this transaction, independent of which tokenId is being queried, while `assetAction` would be different if you are querying one asset vs another.

So for a swap, querying the "to" asset or the "from" asset in "getTransactions" would return different `assetAction` values, but both transactions would have the same `chainAction` with detains about the swap in general.

So, once we fully adopt this new architecture, a lot of this deleted code will end up coming back. I'm not saying we need to change anything now, but just a heads up for the near future.